File: 1-hello-world/python/consumer/receive.py
#!/usr/bin/env python

import pika

# Establish connection to RabbitMQ server on localhost
connection = pika.BlockingConnection()
channel = connection.channel()

# Declare queue (create if not exists)
channel.queue_declare(queue='hello')

# Callback function called when a message is received
def callback(ch, method, properties, body):
    print(" [x] Received %r" % body)

# Register callback function with the queue
channel.basic_consume(callback, queue='hello', no_ack=True)

# Wait for messages
print(' [*] Waiting for messages. To exit press CTRL-C')
channel.start_consuming()

File: 1-hello-world/js/consumer/receive.js
#!/usr/bin/env node

var amqp = require('amqplib/callback_api');

var uri = 'amqp://localhost'
var queueName = 'hello'

// Establish connection to RabbitMQ server
amqp.connect(uri, function(err, conn) {
  conn.createChannel(function(err, ch) {
    // Declare queue (create if not exists)
    ch.assertQueue(queueName, {durable: false});
    // Consume messages from queue
    console.log(" [*] Waiting for messages in %s.", queueName);
    ch.consume(queueName, function(msg) {
      // Callback method for handling messages from queue
      console.log(" [x] Received %s", msg.content.toString());
    }, {noAck: true});
  });
});
